interfaces/apparmor/backend.go

```diff
@@ -105,22 +115,34 @@ func (b *Backend) Initialize(opts *interfaces.SecurityBackendOptions) error {
 	// fail early if this fails for whatever reason.
 	exe, err := os.Readlink(procSelfExe)
 	if err != nil {
-		return fmt.Errorf("cannot read %s: %s", procSelfExe, err)
+		return fmt.Errorf("cannot read %s: %w", procSelfExe, err)
 	}
 
-	if _, err := apparmor_sandbox.SetupSnapConfineSnippets(); err != nil {
+	// TODO:mismatch: regenerate a subset of snippets if needed?
+
+	// snippets are shared between all snap-confine profiles, those from
+	// snapd/core snap and the distro one
+	changed, err := apparmor_sandbox.SetupSnapConfineSnippets()
+	if err != nil {
 		return err
 	}
 
+	if reinitializing && !changed {
+		logger.Noticef("snap-confine snippets not chnaged during reinit, not reloading profiles")
+		return nil
+	}
+
 	// If snapd is executing from the core snap the it means it has
 	// re-executed. In that case we are no longer using the copy of
 	// snap-confine from the host distribution but our own copy. We don't have
 	// to re-compile and load the updated profile as that is performed by
 	// setupSnapConfineReexec below.
```

Comment:
rereading this, "below"  is not very clear,  what is means is really "later during Setup for core/snapd" ?

Reply:
I've rephrased the comment